.github/actions/restore-node/action.yml

```
@@ -123,9 +129,33 @@ runs:
         if test -e ~/endo; then
           scripts/get-packed-versions.sh ~/endo | scripts/resolve-versions.sh
         fi
+        cp package.json package.json.orig
         yarn install
+
+        if cmp -s package.json.orig package.json; then
```

I think you want
```suggestion
        if ! cmp -s package.json.orig package.json; then
```

Otherwise add the comment "if package.json is unchanged...".

---

Thanks.  Good catch!